### PR TITLE
bevy_math: faster sphere sampling

### PR DESCRIPTION
**Pull request by ickk**
Uses fewer transcendental functions than the existing approach

### CODE REVIEW

**Review by mweatherley — APPROVED** [review]
A nice improvement! This is one of our most-used sampling functions as well, so this is pretty valuable.

**Review by IQuick143 — APPROVED** [review]
Overall the change looks like a big improvement.

**Comment by mweatherley** [comment]
Oh, it occurred to me that we should also probably just use this in `Sphere`'s implementation for `sample_interior`. If you don't get around to it in this PR, that's fine though. :)

File: crates/bevy_math/src/sampling/shape_sampling.rs [hunk 1]
```diff
@@ -181,13 +181,13 @@ impl ShapeSample for Sphere {
...
-            y: self.radius * phi.sin() * theta.sin(),
-            z: self.radius * phi.cos(),
-        }
+        let z = rng.gen_range(-1f32..=1f32);
+        let (a_sin, a_cos) = rng.gen_range(-PI..=PI).sin_cos();
+        let c = (1f32 - z * z).sqrt();
+        let x = a_sin * c;
+        let y = a_cos * c;
+
+        Vec3::new(x * self.radius, y * self.radius, z * self.radius)
```
[the remark this answers is not in the document]
Comment (ickk) [reply crates/bevy_math/src/sampling/shape_sampling.rs]: Thanks for the suggestion; I ran a benchmark for this but it didn't end up making a difference greater than noise

run-a
```
sample_boundary/sample_boundary_existing:      [1.8533 ns 1.8539 ns 1.8546 ns]
sample_boundary/sample_boundary_pr:            [824.90 ps 825.58 ps 826.34 ps]
sample_boundary/sample_boundary_pr_suggestion: [829.95 ps 830.40 ps 830.89 ps]
```
run-b
```
sample_boundary/sample_boundary_existing:      [1.8531 ns 1.8539 ns 1.8549 ns]
sample_boundary/sample_boundary_pr:            [831.58 ps 833.63 ps 835.88 ps]
sample_boundary/sample_boundary_pr_suggestion: [826.80 ps 827.36 ps 827.95 ps]
```

I would have done it anyway, except I abstracted the method in order to also improve `sample_interior` as suggested by mweatherley

**Comment by ickk** [comment]
> Oh, it occurred to me that we should also probably just use this in `Sphere`'s implementation for `sample_interior`. If you don't get around to it in this PR, that's fine though. :)

done

**Review by mweatherley — APPROVED** [review]
Nice!

**Review by IQuick143 — APPROVED** [review]

**Pull request merged**